pages/add/index.js

```diff
@@ -3,13 +3,33 @@ import EntriesContainer from "@/components/EntriesContainer";
 import Heading from "@/components/PageHeading";
 import { MainContent } from "@/components/StyledComponents/MainContent";
 import { StyledSection } from "@/components/StyledComponents/StyledSection";
+import { useEffect } from "react";
+import { hasToken } from "@/utils/checkUser";
+
+export async function getServerSideProps(context) {
+  const token = await hasToken(context.req);
+
+  if (!token) {
+    return {
+      redirect: {
+        destination: "/login",
+        permanent: false,
+      },
+    };
+  }
+
+  return { props: {} };
+}
 
 export default function AddEntries({
   handleAddEntry,
   wordList,
   setActivePage,
 }) {
-  setActivePage("add");
+  useEffect(() => {
+    setActivePage("add");
```

Comment:
Usually we'd rather not pass the setters themselves to the components, but rather a handler function that's defined where the setter is initialized.

Reply:
Probably something like `handleActivePage`, but yeah. :)

Reply:
Okay. Thank you. I will add that :)